numpy/linalg/tests/test_linalg.py

```diff
@@ -1915,3 +1915,44 @@ def test_dynamic_programming_logic(self):
     def test_too_few_input_arrays(self):
         assert_raises(ValueError, multi_dot, [])
         assert_raises(ValueError, multi_dot, [np.random.random((3, 3))])
+
+
+class TestTensorinv(object):
+
+    @pytest.mark.parametrize("arr, ind", [
+        (np.random.random((4, 6, 8, 2)), 2),
```

Comment:
Why `random`? Note that this is also not reproducible in case of failure.

Reply:
Hmm, I wonder what the replacement for these sorts of test will be when we drop the requirement of repeatable streams?

You could make generating the test data part of a `setup_class` function with a defined seed.

EDIT: Don't know if that would work with `parametrize` though, there doesn't seem to be a `self` passed along that could be used to access the data.